### PR TITLE
feat: Allow `None` in `skel setBoneValue`

### LINKED ISSUE #1047: `skel.setBoneValue` should accept `None` as value
I've situations where my method get values from outside and it should be okay to provide a (real) value or `None` to clear this value. But currently I've to do
```py
discount_key: db.Key | None
if discount_key is None:
    skel["discount"] = None
else:
    skel.setBoneValue("discount", discount_key)
```
I want to write only
```py
discount_key: db.Key | None
skel.setBoneValue("discount", discount_key)
```

### PR DESCRIPTION
**Pull request by ArneGudermann**
Resolves #1047.
@sveneberth  I hope I have understood your issue correctly.  
This PR sets the value of the bones to `None`.

### CODE REVIEW

**Review by sveneberth — DISMISSED** [review]
Simple solution, thanks

File: src/viur/core/skeleton.py [hunk 1]
```diff
@@ -409,6 +409,9 @@ def setBoneValue(cls, skelValues: t.Any, boneName: str, value: t.Any,
         if not isinstance(bone, BaseBone):
             raise ValueError(f"{boneName} is no valid bone on this skeleton ({skelValues})")
         skelValues[boneName]  # FIXME, ensure this bone is unserialized first
+        if value is None:
+            skelValues[boneName] = None
+            return True
```
Comment (phorward) [inline src/viur/core/skeleton.py:414]: Doesn't deserialization make sense when not being in the `None`-case? 
```suggestion
        if value is None:
            skelValues[boneName] = None
            return True

        _ = skelValues[boneName]  # ensure the bone is being unserialized first
```

**Review by phorward — CHANGES_REQUESTED** [review]

**Review by phorward — CHANGES_REQUESTED** [review]
I'm sorry, but there are two additional issues regarding this PR:

1. What about the `append`-parameter? IMHO the None-logic should only apply when append is False.
2. What about the `language`-parameter? Can you add additional logic to this, too?

**Comment by phorward** [comment]
@ArneGudermann can you please re-check the two points stated?

**Comment by ArneGudermann** [comment]
In #1047 @sveneberth says he want's to clear the bone and I think we should ignore the language and `append`.

**Comment by phorward** [comment]
> In #1047 @sveneberth says he want's to clear the bone and I think we should ignore the language and `append`.

yeah, in his case calling `skel.setBoneValue("bla", None)`, this should be the case. But in `skel.setBoneValue("bla", None, append=True)` removing previously added data is not right, same as `skel.setBoneValue("bla", None, language="en")`, the caller assumes to clear the "en"-part of the bone only.

Please add these cases appropriately.

**Comment by sveneberth** [comment]
I think it's good to only clear the value of a language with `lang`. But append `None` I think is basically nonsense and should raise an exception.

**Review by phorward — CHANGES_REQUESTED** [review]
Hello @ArneGudermann, I'm still a little unhappy with this.

Here's an improved proposal on your PR: https://github.com/ArneGudermann/viur-core/pull/2

This entire function with its ugly name is a mess. `bone.setBoneValue` could also be the place to do it. Nevertheless, I think with my proposal (untested!) this might work better is is more integrative.

In VIUR4, this Gelumpe should be generally revised.

File: src/viur/core/skeleton.py [hunk 2]
```diff
@@ -408,7 +408,16 @@ def setBoneValue(cls, skelValues: t.Any, boneName: str, value: t.Any,
...
         if not isinstance(bone, BaseBone):
             raise ValueError(f"{boneName} is no valid bone on this skeleton ({skelValues})")
-        skelValues[boneName]  # FIXME, ensure this bone is unserialized first
+        if value is None:
+            if append:
+                raise ValueError("Can not append None")
+            if language:
+                skelValues[boneName][language] = None
+            else:
+                skelValues[boneName] = None
```
Comment (sveneberth) [inline src/viur/core/skeleton.py:417]: Shouldn't this better be a list on multiple bones?
```suggestion
                skelValues[boneName][language] = [] if bone.multiple else None
            else:
                skelValues[boneName] = [] if bone.multiple else None
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by phorward — APPROVED** [review]

**Review by sveneberth — APPROVED** [review]

**Pull request merged**